fastlane/lib/fastlane/actions/badge.rb

```diff
@@ -16,7 +16,11 @@ def self.run(params)
           shield_gravity: params[:shield_gravity],
           shield_no_resize: params[:shield_no_resize]
         }
-        Badge::Runner.new.run(params[:path], options)
+        begin
+          Badge::Runner.new.run(params[:path], options)
+        rescue => e
+          UI.user_error!("Something went wrong while running badge: #{e}")
```

Comment:
I think we don't want to re-raise the exception here because we shouldn't be counting this as a 'crash' in our metrics. I can definitely add a comment in there!